src/rgen/Microsoft.Macios.Generator/Extensions/TypeSymbolExtensions.cs

```diff
@@ -222,4 +223,74 @@ public static BindingTypeData<T> GetBindingData<T> (this ISymbol symbol) where T
 		return null;
 	}
 
+	/// <summary>
+	/// Returns if a type is blittable or not.
+	/// </summary>
+	/// <param name="symbol"></param>
+	/// <returns></returns>
+	/// <seealso cref="https://learn.microsoft.com/en-us/dotnet/framework/interop/blittable-and-non-blittable-types"/>
+	public static bool IsBlittable (this ITypeSymbol symbol)
+	{
+		while (true) {
+			// per the documentation, the following system types are blittable
+			switch (symbol.SpecialType) {
+			case SpecialType.System_Byte:
+			case SpecialType.System_SByte:
+			case SpecialType.System_Int16:
+			case SpecialType.System_UInt16:
+			case SpecialType.System_Int32:
+			case SpecialType.System_UInt32:
+			case SpecialType.System_Int64:
+			case SpecialType.System_UInt64:
+			case SpecialType.System_Single:
+			case SpecialType.System_Double:
+			case SpecialType.System_IntPtr:
+			case SpecialType.System_UIntPtr:
+				return true;
+			case SpecialType.System_Array:
+				// if we are dealing with an array, an array of blittable elements is blittable
+				symbol = symbol.ContainingType;
+				continue;
+			}
+
+			if (symbol is IArrayTypeSymbol arrayTypeSymbol) {
+				symbol = arrayTypeSymbol.ElementType;
+				continue;
+			}
+
+			if (symbol.TypeKind == TypeKind.Enum && symbol is INamedTypeSymbol enumSymbol) {
+				// an enum is blittable based on its backing field
+				symbol = enumSymbol.EnumUnderlyingType!;
+				continue;
+			}
+
+			// if we are dealing with a structure, we have to check the layout type and all its children
+			if (symbol.TypeKind == TypeKind.Struct) {
+				// Check for StructLayout attribute with LayoutKind.Sequential
+				var layoutAttribute = symbol.GetAttributes ()
+					.FirstOrDefault (attr => attr.AttributeClass?.ToString () == typeof (StructLayoutAttribute).FullName);
+
+				if (layoutAttribute is not null) {
+					var layoutKind = (LayoutKind) layoutAttribute.ConstructorArguments [0].Value!;
+					if (layoutKind == LayoutKind.Auto) {
+						return false;
+					}
+				} else {
+					return false;
+				}
+
+				// Recursively check all fields of the struct
```

Comment:
Recursively check all _instance_ field of the struct. Static fields don't matter.

Reply:
That is already the case, I have added a test to show that: https://github.com/xamarin/xamarin-macios/pull/21881/commits/71cc0c576aef07b98ed70f7921fb13f704b8c673

Reply:
Just out of curiosity, how would the code to _include_ static fields in the check look like? From what I can see from the documentation, `ITypeSymbol.GetMembers` should return all types, and static fields are also `IFieldSymbol` members.